Lib/glyphsLib/builder/axes.py

```diff
@@ -322,6 +324,51 @@ def font_uses_axis_locations(font):
     )
 
 
+def _potentially_invert_mapping(self, mapping, instance_mapping):
+    # Fontmake introduced an Axis Mapping custom parameter before Glyphs did;
+    # then Glyphs introduced it, but with an inverted interpretation. Then
+    # Glyphs changed the interpretation to match fontmake's. Try and figure out
+    # what's going on.
+    # See https://github.com/googlefonts/glyphsLib/issues/745
+
+    # We want to return a userspace->designspace map for our purposes
+    inverted = {v: k for k, v in mapping.items()}
+    if int(self.font.appVersion) >= 3168:
+        # Mapping is *definitely* designspace->userspace, invert
+        return inverted
```

Comment:
i'm confused, I thought older versions (< 3168?) would interpret it as design->user and thus require flip, whereas newer versions have aligned with fontmake's interpretation and alreadu have user->design? This block seems to contradict my understanding, for it's an inverted mapping if the version is more recent.

Reply:
"Interpret as" is perhaps not helpful here. :-) I just opened Nunito source, converted to Glyphs 3 file format and saved under Glyphs 3310. The custom parameter is *stored as*:

```
wght = {
42 = 200;
61 = 300;
81 = 400;
91 = 500;
101 = 600;
125 = 700;
151 = 800;
178 = 900;
208 = 1000;
};
```

i.e. designspace to userspace.

Reply:
now I am _extremely_ confused

Reply:
well, I for one don't understand this whole mess. It would be nice to have a recap.

Reply:
> I can compile fonts that were not compiling before

good, what I'm worried about are the other fonts that were compiling before and may not after this.
Right now (tested with 3324) on a newly created glyphs3 file, the Axis Mappings is already stored in the source as user->design (or external->internal, as Glyphs calls them), so it is correct and matches fontmake. Wouldn't this PR invert it? I'm seeing 

if int(self.font.appVersion) >= 3168:
    return inverted

Reply:
You are going to love this.

Glyphs now stores user->design for new fonts, but it "upgrades" old fonts by *simply inverting them*. Take your new font, manually edit "appVersion" to (say) 3110, open it in Glyphs and save it again - the mappings will be flipped.

So I guess this means we can't trust the app version number, and we have to probe against the instance mapping.

Reply:
We could have went with my suggestion (back in #745) and made glyphsLib match (old) GlyphsApp behavior and fixed the few fonts that would have been broken by this. Now it is even a bigger mess...

Reply:
I don't deny that it's stupid and wrong and there have may be been questionable choices on both sides, but in 95% of cases the interpretation is completely obvious. Do you have a weight uservalue of 7? You did something wrong. Did all the master locations suddenly fall outside of the design space and you have no valid masters any more? You got the axis mappings backwards. Compare against the instance locations, because we know which way round they are, and if the ranges are wildly different, switch it. It's a mess, but it's not a hard mess to clean up.